Nos/Views/KnownFollowersView.swift

```diff
@@ -8,53 +8,48 @@ struct KnownFollowersView: View {
     @ObservedObject var author: Author
     @Environment(CurrentUser.self) private var currentUser
     
-    var followersRequest: FetchRequest<Follow>
-    var followersResult: FetchedResults<Follow> { followersRequest.wrappedValue }
+    /// The authors that the `source` author follows who also follow the `author`
+    @FetchRequest private var knownFollowers: FetchedResults<Author>
     
-    var followers: Followed {
-        followersResult.map { $0 }
-    }
-    
-    var knownFollowers: [Follow] {
-        followers.filter {
-            guard let source = $0.source else {
-                return false
-            }
-            return source.hasHumanFriendlyName == true &&
-            source != author &&
-            source != currentUser.author &&
-            currentUser.isFollowing(author: source)
-        }
+    /// The authors that will be featured with profile pictures and names
+    var displayedAuthors: ArraySlice<Author> {
+        knownFollowers
+            .filter { $0.hasHumanFriendlyName }
+            .filter { $0.profilePhotoURL != nil }
+            .prefix(3)
     }
     
+    /// The avatars of the authors that will be featured with profile pictures and names
     var avatarURLs: [URL?] {
-        knownFollowers.prefix(3).map { $0.source?.profilePhotoURL }
+        displayedAuthors.compactMap { $0.profilePhotoURL }
     }
     
+    /// The text that will be displayed alongside the avatars listing some of the displayedAuthor's names
     var followText: Text {
         let stringResource: LocalizedStringResource
-        switch avatarURLs.count {
+        let authors = self.displayedAuthors
+        switch authors.count {
+        case 0: 
+            return Text("")
         case 1:
-            guard let name = knownFollowers[safe: 0]?.source?.safeName else {
-                fallthrough
+            guard let name = authors[safe: 0]?.safeName else {
+                return Text("")
             }
             stringResource = LocalizedStringResource.localizable.followedByOne(name)
         case 2:
-            guard let firstName = knownFollowers[safe: 0]?.source?.safeName,
-                let secondName = knownFollowers[safe: 1]?.source?.safeName else {
-                fallthrough
+            guard let firstName = authors[safe: 0]?.safeName,
+                let secondName = authors[safe: 1]?.safeName else {
+                return Text("")
```

Comment:
Maybe this is taking it too far, but couldn't we fall back to the case 1 behavior if there are two authors but we only have one safe name? Similarly for the default case, wouldn't we want to take the first two safe names we can find and use those? Currently it seems like we're just looking at indexes 0 and 1 but it doesn't look like those are any more likely than indexes 2 & 4 to have safe names.

Reply:
The `switch` is looking at the count of `displayedAuthors` which has already been filtered down to authors with safe names, so really the inside of these guards should never be executed (and I put breakpoints in them and never saw them called).

However you could have the situation where say there are 4 `knownFollowers` but only 1 has a name and profile pic, in which case this view would only display "Bob followed you" instead of "Bob and 3 others followed you". Which is not ideal. But that's how it already worked before this PR and I don't think I want to refactor harder to handle that case now.

Reply:
Good points; I'm good with this.